edgelet/edgelet-utils/src/lib.rs

```diff
@@ -190,7 +202,41 @@ mod tests {
             sanitize_dns_label("$a23456789-123456789-123456789-123456789-123456789-123456789-1234")
```

Comment:
By this any chars $, *, \n, etc will be removed and passed with the sanitized output that can be used as object names?

Reply:
Yes, the object names in k8s have some restrictions, and our module names have different restrictions. Rather than rejecting all module names that don't match allowed k8s names, we did what we do for Docker module names, strip out the offending characters, or "sanitize" them. This function was primarily designed to convert "$edgeAgent" to "edgeagent", so sanitizing makes more sense than rejecting names. 

But there were other places where we derive an object name from module settings, and we used the same function, but other objects likes PVCs are less restrictive in their names. We need to sanitize this names differently.